src/plugins/vis_type_metric/public/to_ast.ts

```diff
@@ -55,7 +55,10 @@ export const toExpressionAst: VisToExpressionAst<VisParams> = (vis, params) => {
   // fix formatter for percentage mode
   if (get(vis.params, 'metric.percentageMode') === true) {
     schemas.metric.forEach((metric: SchemaConfig) => {
-      metric.format = { id: 'percent' };
+      metric.format = {
+        id: 'percent',
+        params: { pattern: get(vis.params, 'metric.percentageFormatPattern', undefined) },
```

Comment:
nit:
```suggestion
        params: { pattern: percentageFormatPattern },
```
where `percentageFormatPattern` could be extracted from `vis.params.metric` at line 52

Reply:
done